databricks/koalas/indexes.py

```diff
@@ -60,9 +60,17 @@ class Index(IndexOpsMixin):
 
     >>> ks.DataFrame({'a': [1, 2, 3]}, index=list('abc')).index
     Index(['a', 'b', 'c'], dtype='object')
+
+    >>> Index([1, 2, 3])
+    Int64Index([1, 2, 3], dtype='int64')
+
+    >>> Index(list('abc'))
+    Index(['a', 'b', 'c'], dtype='object')
     """
 
-    def __init__(self, kdf: DataFrame, scol: Optional[spark.Column] = None) -> None:
+    def __init__(self, kdf_or_names: Union[DataFrame, list],
+                 scol: Optional[spark.Column] = None) -> None:
+        kdf = DataFrame(index=kdf_or_names) if isinstance(kdf_or_names, list) else kdf_or_names
```

Comment:
ah, yes i understand, i'm going to add them right away!
And always thanks for your time to review my works. 😸

Reply:
I have a question!

if we apply this logic, we have some problem with the existing codes that create index with `scol` like:

```python
return Index(self._kdf, scol)
```

will occurs `AssertionError` because the second parameter is `dtype`, not `scol`

```python
>>> kdf.index
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/indexes.py", line 473, in __repr__
    pindex = self._kdf.head(max_display_count + 1).index._with_new_scol(self._scol).to_pandas()
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/indexes.py", line 95, in _with_new_scol
    return Index(self._kdf, scol)
  File "/Users/haejoonlee/Desktop/git_store/koalas/databricks/koalas/indexes.py", line 74, in __init__
    assert dtype is None
```

so i think we should some fix to consider these codes,

and i suggest two solutions:

one is using the keyword explicitly where function is called like:

```python
return Index(self._kdf, scol=scol)  # rather than return Index(self._kdf, scol)
```

the other is arranging parameter order to `scol` to be a second parameter like.

```python
def __init__(self, data: Union[DataFrame, list], 
                   scol: Optional[spark.Column] = None, dtype=None, name=None) -> None:
```

i'm not pretty sure which solution is better, what are you think??

Reply:
hmm.. maybe i think the second solution is better,

otherwise we should fix whole code in koalas that using `scol` parameter without explicitly using?

Reply:
We should take the first solution. We should place our own parameters after pandas' parameters.

Reply:
ah! i got it, i'll fix them.